src/arithmetic/arithmetic_expression_construct.c

```diff
@@ -411,31 +426,49 @@ static AR_ExpNode *_AR_ExpFromMapProjection(const cypher_astnode_t *expr) {
 			// { .name }
 			prop = cypher_ast_map_projection_property_get_prop_name(selector);
 			prop_name = cypher_ast_prop_name_get_value(prop);
-			children[i * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
+			children[j * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
 			AR_ExpNode *entity = AR_EXP_NewVariableOperandNode(entity_name);
-			children[i * 2 + 1] = AR_EXP_NewAttributeAccessNode(entity, prop_name);
+			children[j * 2 + 1] = AR_EXP_NewAttributeAccessNode(entity, prop_name);
+			j++;
 		} else if(t == CYPHER_AST_MAP_PROJECTION_LITERAL) {
 			// { v: n.v }
 			prop = cypher_ast_map_projection_literal_get_prop_name(selector);
 			prop_name = cypher_ast_prop_name_get_value(prop);
 			const cypher_astnode_t *literal_exp =
 				cypher_ast_map_projection_literal_get_expression(selector);
-			children[i * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
-			children[i * 2 + 1] = AR_EXP_FromASTNode(literal_exp);
+			children[j * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
+			children[j * 2 + 1] = AR_EXP_FromASTNode(literal_exp);
+			j++;
 		} else if(t == CYPHER_AST_MAP_PROJECTION_IDENTIFIER) {
 			// { v }
 			prop = cypher_ast_map_projection_identifier_get_identifier(selector);
 			prop_name = cypher_ast_identifier_get_name(prop);
-
-			children[i * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
-
-			children[i * 2 + 1] = AR_EXP_NewVariableOperandNode(prop_name);
+			children[j * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
+			children[j * 2 + 1] = AR_EXP_NewVariableOperandNode(prop_name);
+			j++;
+		} else if(t == CYPHER_AST_MAP_PROJECTION_ALL_PROPERTIES) {
+			// { .* }
+			// Use properties() to get a map with all properties
+			propertiesOp = AR_EXP_NewOpNode("properties", true, 1);
+			AR_ExpNode **propertiesChildren = propertiesOp->op.children;
+			propertiesChildren[0] = AR_EXP_NewVariableOperandNode(entity_name);
 		} else {
 			ASSERT("Unexpected AST node type" && false);
 		}
 	}
 
-	return op;
+	if(propertiesOp && tomapOp) {
+		// To support case like: CREATE (a:A {z:1}) RETURN a{.*, .undefinedProp}
+		AR_ExpNode *opAddMap = AR_EXP_NewOpNode("addmap", true, 2);
+		AR_ExpNode **childrenAddMapp = opAddMap->op.children;
+		childrenAddMapp[0] = tomapOp;
+		childrenAddMapp[1] = propertiesOp;
+		return opAddMap;
+	} else if (propertiesOp) {
+		return propertiesOp;
```

Comment:
I guess it is possible to run the following:
`MATCH (a) RETURN a {.*}`
We should return `propertiesOp` but note `tomapOp` was already allocated, so it is leaked, need better handling for this case.